PlaywrigthUITests/PageObjects/DemoQAWebTablesPage.cs

```diff
@@ -87,18 +93,124 @@ public async Task VerifyTableRowContent(string headerName = "First Name", string
+        Assert.That(await EmailField.GetAttributeAsync("value") == email, $"The Email field was not filled with: '{email}'.");
+    }
+    
+    public async Task FillAge (string age)
+    {
+        if(age == null) throw new ArgumentException("Age field should not be null.");
+        await AgeField.FillAsync(age);
+        Assert.That(await AgeField.GetAttributeAsync("value") == age, $"The Age field was not filled with: '{age}'.");
+    }
+    
+    public async Task FillSalary (string salary)
+    {
+        if(salary == null) throw new ArgumentException("Salary field should not be null.");
+        await SalaryField.FillAsync(salary);
+        Assert.That(await SalaryField.GetAttributeAsync("value") == salary, $"The Salary field was not filled with: '{salary}'.");
+    }
+    
+    public async Task FillDepartment (string department)
+    {
+        if(department == null) throw new ArgumentException("Department field should not be null.");
+        await DepartmentField.FillAsync(department);
+        Assert.That(await DepartmentField.GetAttributeAsync("value") == department, $"The Department field was not filled with: '{department}'.");
+    }
 
-}
+    public async Task ClickSubmit()
+    {
+        var submit = Page.GetByRole(AriaRole.Button, new() { Name = "Submit" });
+        await submit.ClickAsync();
+    }
+
+    public async Task<bool> FindRowInTable(string firstName, string lastName, string email, string age, string salary, string department)
+    {
+        if (string.IsNullOrWhiteSpace(firstName) ||
+            string.IsNullOrWhiteSpace(lastName) ||
+            string.IsNullOrWhiteSpace(email) ||
+            string.IsNullOrWhiteSpace(age) ||
+            string.IsNullOrWhiteSpace(salary) ||
+            string.IsNullOrWhiteSpace(department))
+        {
+            throw new ArgumentException("One or more fields are empty.");
+        }
+
+        var rows = await Page.Locator("//div[@class=\"rt-tr-group\"]").AllAsync();
+
+        if (rows == null || rows.Count == 0)
+        {
+            throw new ArgumentException("No rows found in the table.");
+        }
+
+        bool rowFound = false;
+
+        foreach (var row in rows)
+        {
+            var rowText = await row.InnerTextAsync();
+
+            if (rowText.Contains(firstName) &&
+                rowText.Contains(lastName) &&
+                rowText.Contains(email) &&
+                rowText.Contains(age) &&
+                rowText.Contains(salary) &&
+                rowText.Contains(department))
+            {
+                rowFound = true;
+                break;
+            }
+        }
+        return rowFound;
+    }
+    
+    public async Task ClickEditButton(string email)
+    {
+        if(string.IsNullOrWhiteSpace(email)) throw new ArgumentException("Email field is empty.");
+        var removeRowButton =
+            Page.Locator($"//*[@class=\"rt-tr-group\"]//*[contains(text(),'{email}')]/following-sibling::div/*/span[@title=\"Edit\"]");
```

Comment:
this is well done!!! Most complex locators which we learned.
The only instead of
following-sibling::div/*/
use
following-sibling::div//
in this case you are not strictly bound to the structure div/some element/span. Instead of some element (div) can be div/div or div/div/div and selector will still work

Reply:
thanks, updated